src/Reports/SubsiteReportWrapper.php

```diff
@@ -64,13 +65,17 @@ public function beforeQuery($params)
     {
         // The user has select a few specific sites
         if (!empty($params['Subsites'])) {
-            Subsite::$force_subsite = $params['Subsites'];
+            SubsiteState::singleton()->withState(function (SubsiteState $newState) use ($params) {
```

Comment:
The previous logic with presumably setting the subsite to use for the rest of the request?

This block is doing absolutely nothing?  Calling `$newState->setSubsiteID()` within `withState` set the subsiteID for the duration of the block, not the rest of the request

The comment also applies to the withState() calls below in this PR

Reply:
I also think that these two methods do nothing. Should we mark them as deprecated and replace with new methods that we invoke callback function.
```php
    public function beforeAfterQuery($callback, $args, $params)
    {
        return SubsiteState::singleton()->withState(function (SubsiteState $newState) use ($callback, $args, $params) {

            if (!empty($params['Subsites'])) {
                $newState->setSubsiteId($params['Subsites']);
            } else {
                $subsites = Subsite::accessible_sites('CMS_ACCESS_CMSMain');
                $options = $subsites->toDropdownMap('ID', 'Title');
                $newState->setSubsiteId(join(',', array_keys($options ?? [])));
            }

            $result = $callback($args);
            $newState->setSubsiteId(null);

            return $result;
        });
    }
```

Reply:
We cannot create a new `beforeAfterQuery()` method, because SubsiteReportWrapper extends the abstract class ReportWrapper which has

```
    /**
     * Override this method to perform some actions prior to querying.
     */
    public function beforeQuery($params)
    {
    }

    /**
     * Override this method to perform some actions after querying.
     */
    public function afterQuery()
    {
    }
```

However looking closer at SubsiteReportWrapper and ReportWrapper, I don't think there's anyway we can replicate the old functionality.

There's also no reference in the codebase to SubsiteReportWrapper, so I'm inclined to just deprecate SubsiteReportWrapper in CMS4 and delete in CMS 5